contracts/protocol/facets/ProtocolInitializationHandlerFacet.sol

```diff
@@ -24,6 +26,15 @@ contract ProtocolInitializationHandlerFacet is IBosonProtocolInitializationHandl
         _;
     }
 
+    /**
+     * @notice Constructor
+     *
+     * @dev This constructor is used to prevent invocation of initialize directly on deployed contract.
+     */
+    constructor() {
+        thisAddress = address(this);
+    }
+
     /**
      * @notice Initializes the protocol after the deployment.
      * This function is callable only once for each version
```

Comment:
I have noticed that in some other places we have put the conditions in the comments, this is an external method, worth doing it here ?

Reply:
I added them.

Maybe initialize are slightly different than other external methods, since they are not part of the interface, and they don't go into documentation. But it make sense to put the conditions here for other devs.